lib/ItemList.cpp

```diff
@@ -33,32 +33,52 @@ namespace tpublic
 	{
 		if(aItemData->IsUnique() && HasItems(aItemInstance.m_itemId, 1))
 			return false;
-		
+
+		// Check we can actually add all the items to the inventory
+		if(!CanAddMultipleToInventory(aItemInstance.m_itemId, aItemInstance.m_quantity, aItemData, aSize))
+			return false;
+
+		uint32_t remaining = aItemInstance.m_quantity;
+		// 1st inv pass, attempt to stack existing item stacks
+		for(uint32_t i = 0; i < (uint32_t)m_entries.size() && i < aSize; i++)
+        {
+        	Entry& t = m_entries[i];
+
+        	if(t.m_item.IsSet() && t.m_item.m_itemId == aItemInstance.m_itemId)
+			{
+				assert(t.m_item.m_itemId == aItemData->m_id);
+
+				if(t.m_item.m_quantity < aItemData->m_stackSize)
+				{
+                	uint32_t t_amountToAdd = Base::Min(aItemData->m_stackSize - t.m_item.m_quantity, remaining);
+					t.m_item.m_quantity += t_amountToAdd;
+					remaining -= t_amountToAdd;
+					m_version++;
```

Comment:
@demogorgon1 I was not sure how your versioning works, so I update the version every single time an inventory edit is made. If the version should only change once per transaction, the m_version++ could be moved into the if statement two lines down.